### ISSUE TITLE
Possible decrease in image quality after upgrade

### ISSUE THREAD

**Issue by alexisrolland** [damian0815/compel#27]
Hi Damian,

Me again for a feedback. I have upgraded compel from `compel==0.1.10` to `compel==1.1.3`. I noticed some important changes on the generated images, feeling a bit like a drop in quality actually. In particular on faces generation (not very visible in the thumbnails but you can open the images in a dedicated tab). I'm not sure if it's under control or some undesired side effect from the changes you did. I thought I would let you know...

Example:

- Diffusers `0.14.0`
- Prompt: (armor, beautiful woman, bathing, full body)+(8k, RAW photo, highest quality, masterpiece, High detail RAW color photo, professional photo, realistic, photo realism, highest quality, best shadow, best illustration, ultra-high resolution, physics-based rendering, cinematic lighting)+ 
- Negative Prompt: (CGI, 3d, lowres, text, error, cropped, worst quality, low quality, jpeg artifacts, ugly, duplicate, morbid, mutilated, out of frame, extra fingers, mutated hands, poorly drawn hands, poorly drawn face, mutation, deformed, blurry, dehydrated, bad anatomy, bad proportions, extra limbs, cloned face, disfigured, gross proportions, malformed limbs, missing arms, missing legs, extra arms, extra legs, fused fingers, too many fingers, long neck, username, watermark, signature)++ 
- Seed: -3058875626 
- Resolution: 9:16 (576x1024)
- Steps: 50
- Guidance: 5
- Scheduler: UNIPC
- Model: Lykon/DreamShaper

| `compel==0.1.10` | `compel==1.1.3` |
| --- | --- |
| ![0 1 10-torch](https://user-images.githubusercontent.com/13064696/232800195-c0f944a2-2b92-49ea-93d9-da8f7323fc43.png) | ![b01e4ea2-b879-4176-9161-88260db131f3](https://user-images.githubusercontent.com/13064696/233033336-469a39e8-b5b5-4d11-b4af-6bc35f36c30d.png) |

**Comment by alexisrolland** [comment]
Something strange happened, I rolled back my code to `compel==0.1.10` and `torch.cat([compel.build_conditioning_tensor(prompt) for prompt in prompts])` and I am not able to regenerate the first picture in the top left corner. I always get the picture in the bottom left corner with `compel==0.1.10`, whether I use the callable `compel(prompts)` or `torch.cat([compel.build_conditioning_tensor(prompt) for prompt in prompts])`...

**Comment by alexisrolland** [comment]
I updated my tests results in the post above. There was another change on my side that introduced some side effect. Now we can clearly see the difference between the two versions.

**Comment by damian0815** [comment]
interesting. would you mind doing a comparison with `compel==1.1.0` and `compel==1.0.0` ? i don't have a setup that can readily replicate your test conditions.

also, is the problem apparent with other prompts? i can see that the diffusion process is *changed*, i would hesitate to agree that there's less *quality* - i mean i can see the face is drawn worse in the profile angle but i'd rather attribute that to that model just generally not being as good at drawing faces in profile than in frontal view.

**Comment by alexisrolland** [comment]
Please ignore... this seemed to be an issue on my side.
Thanks

**Comment by damian0815** [comment]
ahha all good :)

**Comment by alexisrolland** [comment]
For the record, this issue was because we introduced a function to convert A1111 syntax with () and [] to Compel syntax. The problem is when user provides a prompt that already has the Compel syntax, then a +/- gets added for the pair of brackets that's already...

This syntax conversion is way less trivial than it looks. If at some point you consider working on this I'm sure you could do a better job than I did.

**Comment by damian0815** [comment]
there is a conversion script floating around...

here it is https://gist.github.com/watzon/1e15172550a537532b0b52e12703aada

i am waiting for somebody to volunteer to convert this to python or something, and offer to maintain it long term.

**Comment by alexisrolland** [comment]
Thanks for sharing, after testing it seems this script also has the same limitation as the one we've got... when a prompt already contains Compel syntax with `()+` and `()-`, it would still add an extra `+` and `-` which I think is a problem.

**Comment by alexisrolland** [comment]
> maybe you can filter prompts for )+ or )-

That's what I did! Thanks
